library/std/src/thread/mod.rs

```diff
@@ -718,7 +720,25 @@ pub(crate) fn set_current(thread: Thread) {
 /// In contrast to the public `current` function, this will not panic if called
 /// from inside a TLS destructor.
 pub(crate) fn try_current() -> Option<Thread> {
-    CURRENT.try_with(|current| current.get_or_init(|| Thread::new_unnamed()).clone()).ok()
+    CURRENT
+        .try_with(|current| {
+            let thread = current.get_or_init(Thread::new_unnamed).clone();
+            CURRENT_ID.set(Some(thread.id()));
```

Comment:
So this does a `set` over and over again? Isn't that excessive?

Reply:
Oh right, not sure how I didn't catch that. I suppose I can just move the `CURRENT_ID.set()` into the `get_or_init()` so it's only executed once.